src/Type/PostObject/Mutation/PostObjectCreate.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace WPGraphQL\Type\PostObject\Mutation;
+
+use GraphQLRelay\Relay;
+use WPGraphQL\Type\PostObject\PostObjectMutation;
+use WPGraphQL\Types;
+
+/**
+ * Class PostObjectCreate
+ *
+ * @package WPGraphQL\Type\PostObject\Mutation
+ */
+class PostObjectCreate {
+
+	/**
+	 * Holds the mutation field definition
+	 *
+	 * @var array $mutation
+	 */
+	private static $mutation;
+
+	/**
+	 * Defines the create mutation for PostTypeObjects
+	 *
+	 * @param \WP_Post_Type $post_type_object
+	 *
+	 * @return array|mixed
+	 */
+	public static function mutate( \WP_Post_Type $post_type_object ) {
+
+		if ( null === self::$mutation ) {
+			self::$mutation = [];
+		}
+
+		if ( ! empty( $post_type_object->graphql_single_name ) && empty( self::$mutation[ $post_type_object->graphql_single_name ] ) ) :
```

Comment:
so this is actually the definition of the schema (what mutations are allowed to be executed), not the execution of the mutation.

So, this ensures that we only defined this create mutation exactly once for this post_type, and any circular references will use the existing definition instead of creating a new definition for it.